velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -1383,6 +1385,73 @@ TpchPlan TpchQueryBuilder::getQ16Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ17Plan() const {
+  std::vector<std::string> lineitemColumns = {
+      "l_partkey", "l_extendedprice", "l_quantity"};
+  std::vector<std::string> partColumns = {
+      "p_partkey", "p_brand", "p_container"};
+
+  auto lineitemSelectedRowType = getRowType(kLineitem, lineitemColumns);
+  const auto& lineitemFileColumns = getFileColumnNames(kLineitem);
+  auto partSelectedRowType = getRowType(kPart, partColumns);
+  const auto& partFileColumns = getFileColumnNames(kPart);
+
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+  core::PlanNodeId lineitemScanNodeId;
+  core::PlanNodeId lineitemScanNodeIdSubQuery;
+  core::PlanNodeId partScanNodeId;
+
+  auto part = PlanBuilder(planNodeIdGenerator)
+                  .tableScan(
+                      kPart,
+                      partSelectedRowType,
+                      partFileColumns,
+                      {"p_brand = 'Brand#23'", "p_container = 'MED BOX'"})
+                  .capturePlanNodeId(partScanNodeId)
+                  .planNode();
+
+  auto lineitemJoinPart =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kLineitem, lineitemSelectedRowType, lineitemFileColumns)
+          .capturePlanNodeId(lineitemScanNodeId)
+          .hashJoin(
+              {"l_partkey"},
+              {"p_partkey"},
+              part,
+              "",
+              {"l_quantity", "p_partkey", "l_extendedprice"})
+          .planNode();
+
+  auto plan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kLineitem, lineitemSelectedRowType, lineitemFileColumns)
+          .capturePlanNodeId(lineitemScanNodeIdSubQuery)
+          .partialAggregation({"l_partkey"}, {"avg(l_quantity) as avg_"})
+          .localPartition({})
```

Comment:
This makes the processing go single-threaded. You should instead partition on l_partkey

```
.localPartition({"l_partkey"})
```

Reply:
Thanks for the suggestion, updated